### PR TITLE
Sample with no signal should now fail

### PR DESCRIPTION
**Pull request by johandahlberg**
No signal, i.e. HEX=0 and VIC=0 is ok for the negative control sample, but not for any other type of sample. This handles that.

### CODE REVIEW

**Comment by johandahlberg** [comment]
Do not merge this yet! I discovered one thing.

**Comment by johandahlberg** [comment]
Ok, I fixed it. It is alright to review now.

File: clarity-ext-scripts/tests/test_rtpcr_analysis_service.py [hunk 1]
```diff
@@ -204,3 +204,66 @@ def test_can_analyze_samples_scenario6(self):
...
+            {"id": "neg-2", "diagnosis_result": COVID_RESPONSE_NEGATIVE},
+            {"id": "cov-1",
+                "diagnosis_result": COVID_RESPONSE_POSITIVE},
+            {"id": "cov-2",
+                "diagnosis_result": COVID_RESPONSE_POSITIVE},
+            {"id": "cov-3", "diagnosis_result": FAILED_BY_TOO_HIGH_COVID_VALUE}]
+
+        assert expected == list(result)
+
+    def test_can_analyze_samples_scenario8(self):
```
Comment (GitEdvard) [inline clarity-ext-scripts/tests/test_rtpcr_analysis_service.py:240]: Perhaps there should be a test with positive control have fam = 0 and hex =0, and all samples are failed?

File: clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analysis_service.py [hunk 2]
```diff
@@ -64,18 +64,28 @@ def __init__(self, covid_reporter_key, internal_control_reporter_key):
...
 
-    def _analyze_sample(self, sample):
+    # POS Control
+    # FAM=0 and VIC=0 => FAIL
+    # NEG Control
+    # FAM=0 and VIC=0 => NEG
+    # SAMPLE
+    # FAM=0 and VIC=0 => FAIL
+
+    def _analyze_sample(self, sample, is_neg_control=False):
```
Comment (GitEdvard) [inline clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analysis_service.py:74]: I don't like to have is_neg_control as an input arg here. But this doesn't matter today, as this fixes the thing.

File: clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analysis_service.py [hunk 3]
```diff
@@ -64,18 +64,28 @@ def __init__(self, covid_reporter_key, internal_control_reporter_key):
...
+            else:
+                return FAILED_BY_INTERNAL_CONTROL
         elif covid_ct == 0 and internal_control_ct <= self.INTERNAL_CONTROL_THRESHOLD:
             return COVID_RESPONSE_NEGATIVE
         elif covid_ct == 0 and internal_control_ct > self.INTERNAL_CONTROL_THRESHOLD:
             return FAILED_BY_INTERNAL_CONTROL
         elif covid_ct > self.COVID_CONTROL_THRESHOLD:
             return FAILED_BY_TOO_HIGH_COVID_VALUE
-        elif covid_ct <= self.COVID_CONTROL_THRESHOLD:
+        elif covid_ct <= self.COVID_CONTROL_THRESHOLD and covid_ct > 0:
```
Comment (withrocks) [inline clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analysis_service.py:88]: I'd suggest to have the lower bound on the left, perhaps (but not necessarily) using Python's syntactic sugar:

`elif 0 < covid_ct <= self.COVID_CONTROL_THRESHOLD`

File: clarity-ext-scripts/tests/test_rtpcr_analysis_service.py (hunk 1 above)
Comment (johandahlberg) [reply clarity-ext-scripts/tests/test_rtpcr_analysis_service.py:240]: Will add one.

File: clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analysis_service.py (hunk 3 above)
Comment (johandahlberg) [reply clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analysis_service.py:88]: Will fix.

File: clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analysis_service.py [hunk 4]
```diff
@@ -64,18 +64,28 @@ def __init__(self, covid_reporter_key, internal_control_reporter_key):
 
     # TODO Later we might have to account for giving two trails over 38 as positive.
 
-    def _analyze_sample(self, sample):
+    # POS Control
```
Comment (withrocks) [inline clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analysis_service.py:67]: I think we might want to merge these comments with those above (original)?

Comment (johandahlberg) [reply clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analysis_service.py:67]: Yeah, lets write some proper docs for this thing at a later stage.

Comment (johandahlberg) [reply clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analysis_service.py:67]: I think it you think it is ok we merge it.

**Pull request merged**